tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -1127,27 +1127,160 @@ public function test_loading_strategy_with_all_defer_dependencies() {
 	 *
 	 * @covers WP_Scripts::do_item
 	 * @covers WP_Scripts::get_eligible_loading_strategy
+	 * @covers ::wp_register_script
 	 * @covers ::wp_enqueue_script
 	 */
 	public function test_defer_with_async_dependent() {
 		// case with one async dependent.
-		wp_enqueue_script( 'main-script-d4', '/main-script-d4.js', array(), null, array( 'strategy' => 'defer' ) );
-		wp_enqueue_script( 'dependent-script-d4-1', '/dependent-script-d4-1.js', array( 'main-script-d4' ), null, array( 'strategy' => 'defer' ) );
-		wp_enqueue_script( 'dependent-script-d4-2', '/dependent-script-d4-2.js', array( 'dependent-script-d4-1' ), null, array( 'strategy' => 'async' ) );
-		wp_enqueue_script( 'dependent-script-d4-3', '/dependent-script-d4-3.js', array( 'dependent-script-d4-2' ), null, array( 'strategy' => 'defer' ) );
+		wp_register_script( 'main-script-d4', '/main-script-d4.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script(
+			'dependent-script-d4-1',
+			'/dependent-script-d4-1.js',
+			array( 'main-script-d4' ),
+			null,
+			array(
+				'strategy'      => 'defer',
+				'fetchpriority' => 'auto',
+			)
+		);
+		wp_enqueue_script(
+			'dependent-script-d4-2',
+			'/dependent-script-d4-2.js',
+			array( 'dependent-script-d4-1' ),
+			null,
+			array(
+				'strategy'      => 'async',
+				'fetchpriority' => 'low',
+			)
+		);
+		wp_enqueue_script(
+			'dependent-script-d4-3',
+			'/dependent-script-d4-3.js',
+			array( 'dependent-script-d4-2' ),
+			null,
+			array(
+				'strategy'      => 'defer',
+				'fetchpriority' => 'high',
+			)
+		);
 		$output    = get_echo( 'wp_print_scripts' );
 		$expected  = "<script type='text/javascript' src='/main-script-d4.js' id='main-script-d4-js' defer='defer' data-wp-strategy='defer'></script>\n";
 		$expected .= "<script type='text/javascript' src='/dependent-script-d4-1.js' id='dependent-script-d4-1-js' defer='defer' data-wp-strategy='defer'></script>\n";
-		$expected .= "<script type='text/javascript' src='/dependent-script-d4-2.js' id='dependent-script-d4-2-js' defer='defer' data-wp-strategy='async'></script>\n";
-		$expected .= "<script type='text/javascript' src='/dependent-script-d4-3.js' id='dependent-script-d4-3-js' defer='defer' data-wp-strategy='defer'></script>\n";
+		$expected .= "<script type='text/javascript' src='/dependent-script-d4-2.js' id='dependent-script-d4-2-js' defer='defer' data-wp-strategy='async' fetchpriority='low'></script>\n";
+		$expected .= "<script type='text/javascript' src='/dependent-script-d4-3.js' id='dependent-script-d4-3-js' defer='defer' data-wp-strategy='defer' fetchpriority='high'></script>\n";
 
 		$this->assertEqualHTML( $expected, $output, '<body>', 'Scripts registered as defer but that have dependents that are async are expected to have said dependents deferred.' );
 	}
 
+	/**
+	 * Data provider for test_fetchpriority_values.
+	 *
+	 * @return array<string, array{fetchpriority: string}>
+	 */
+	public function data_provider_fetchpriority_values(): array {
```

Comment:
I believe it's encouraged to use `static` data providers ([in phpunit 10 they _must_ be static](https://docs.phpunit.de/en/10.5/writing-tests-for-phpunit.html#data-providers)):

```suggestion
	public static function data_provider_fetchpriority_values(): array {
```